luigi_pipeline/seqr_loading.py

```diff
@@ -1,16 +1,76 @@
+import logging
+
 import luigi
 
 from lib.hail_tasks import HailMatrixTableTask, HailElasticSearchTask, GCSorLocalTarget
+from lib.global_config import GlobalConfig
+
+logger = logging.getLogger(__name__)
 
 
+class SeqrValidationError(Exception):
+    pass
+
 class SeqrVCFToMTTask(HailMatrixTableTask):
     """
     Inherits from a Hail MT Class to get helper function logic. Main logic to do annotations here.
     """
     reference_mt_path = luigi.Parameter(description='Path to the matrix table storing the reference variants.')
+    sample_type = luigi.Parameter(description='Sample type, WGS or WES')
+    validate = luigi.BoolParameter(default=True, description='Perform validation on the dataset.')
+
+    @staticmethod
+    def validate_mt(mt, genome_version, sample_type):
+        """
+        Validate the mt by checking against a list of common coding and non-coding variants given its
+        genome version. This validates genome_version, variants, and the reported sample type.
+
+        :param mt: mt to validate
+        :param genome_version: reference genome version
+        :param sample_type: WGS or WES
+        :return: True or Exception
+        """
+        imputed_stats = HailMatrixTableTask.hl_mt_impute_sample_type(mt, genome_version)
+
+        for name, stat in imputed_stats.items():
+            logger.info('Table contains %i out of %i common %s variants.' %
+                        (stat['matched_count'], stat['total_count'], name))
+
+        has_coding = imputed_stats['coding']['match']
+        has_noncoding = imputed_stats['noncoding']['match']
+
+        if not has_coding and not has_noncoding:
+            # No common variants detected.
+            raise SeqrValidationError(
+                'Genome version validation error: dataset specified as GRCh{genome_version} but doesn\'t contain '
+                'the expected number of common GRCh{genome_version} variants'.format(genome_version=genome_version)
+            )
+        elif has_noncoding and not has_coding:
+            # Non coding only.
+            raise SeqrValidationError(
+                'Dataset is contains noncoding variants but is missing common coding variants for'
```

Comment:
Should this be "Dataset contains" instead of "Dataset is contains"? Also, this error message is not consistent with the others. It does not say "___ validation error" and it is not clear why this is a problem or how it can be corrected. The others have a definite path to resolution: specify the correct reference genome version or sample type.

Reply:
Hmm, I just modified https://github.com/macarthur-lab/hail-elasticsearch-pipelines/blob/master/hail_scripts/v01/utils/validate_vds.py#L61. I think this is just an error on the dataset, but can be more descriptive ('Please verify the dataset is valid.') But I can't think of much else besides that?

Reply:
Fair enough. I was thinking that, depending on how errors are reported, this could look like an information message after which the program stops.